packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -63,9 +63,9 @@ List<Directory> _directoryListFromEnvironment(
   }).toList();
 }
 
-Directory _directoryFromEnvironment(String envVar, String fallback) {
-  assert(envVar != null);
-  final String value = _getenv(envVar);
+Directory? _directoryFromEnvironment(String envVar, String? fallback) {
```

Comment:
I believe `null` is always passed to `fallback` for this method. You should be able to just remove the `fallback` parameter and just return `null` if `value == null || value.isEmpty`.

Also, the name of this method is close to the name of the other method `_directoryFromEnv` which may cause confusion when reading. The name of the two methods could be something like `_directoryFromEnv` and `directoryFromEnvWithFallback`. 

@stuartmorgan may prefer something different

Reply:
Done